File: food_ecommerce_backend/lib/data/models/product.dart
// Plain immutable Product model without freezed/generator dependency.

class Product {
  final String id;
  final String title;
  final String? description;
  final double price;
  final String? category;
  final String? imageUrl;
  final int? stock;

  const Product({
    required this.id,
    required this.title,
    this.description,
    required this.price,
    this.category,
    this.imageUrl,
    this.stock,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['id'] as String,
        title: json['title'] as String,
        description: json['description'] as String?,
        price: (json['price'] as num).toDouble(),
        category: json['category'] as String?,
        imageUrl: json['imageUrl'] as String?,
        stock: json['stock'] as int?,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'title': title,
        'description': description,
        'price': price,
        'category': category,
        'imageUrl': imageUrl,
        'stock': stock,
      };
}

File: food_ecommerce_backend/lib/backend/src/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  title: String,
  description: String,
  price: Number,
  category: String,
  imageUrl: String,
  rating: {type: Number, default: 0},
  stock: {type: Number, default: 1},
  createdAt: {type: Date, default: Date.now}
});

module.exports = mongoose.model('Product', productSchema);
